src/components/BlockchainList/PostsList.tsx

```diff
@@ -19,44 +21,56 @@ const scrollContainer = classnames(
 )
 
 function BlockchainPostsListSuspended() {
-  const { selectedPosts } = useSnapshot(PostStore)
+  const { selectedPosts, selectedToken, postsAmount, limit } =
+    useSnapshot(PostStore)
   const { selectedType } = useSnapshot(SelectedTypeStore)
+  const totalPosts = postsAmount[selectedType]
   const { hashStore, hashId } = useHashParams()
   const matchStore = hashStore && hashStore === selectedType
+  const [scrolledLimit, setScrolledLimit] = useState(limit)
 
-  const sliceToSpecificPost = (totalPosts: number) => {
-    if (!(matchStore || hashId)) return 10
-    return totalPosts - Number(hashId)
-  }
+  useEffect(() => {
+    function updateLimitIfNeeded() {
+      if (!(matchStore || hashId)) return
+      const neededLimit = totalPosts - Number(hashId)
+      setScrolledLimit(Math.max(neededLimit, PostStore.limit))
+      window.scrollTo({ top: window.innerHeight, behavior: 'smooth' })
+    }
 
-  const posts = PostStore.selectedToken
+    void updateLimitIfNeeded()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const posts = selectedToken
     ? selectedPosts.filter(
         ({ derivativeAddress }) => derivativeAddress === PostStore.selectedToken
       )
     : selectedPosts
-  const [loadedPosts, setLoadedPosts] = useState(
-    posts.slice(0, sliceToSpecificPost(posts.length))
-  )
-  const loadedItemsAmount = loadedPosts.length
-  const loadMoreItems = () => {
-    setLoadedPosts([
-      ...loadedPosts,
-      ...posts.slice(loadedItemsAmount, loadedItemsAmount + 10),
-    ])
-  }
 
-  if (matchStore && hashId) useScrollToAnchor({ callback: flashingPost })
+  if (matchStore && hashId && selectedPosts.length)
+    useScrollToAnchor({ callback: flashingPost })
 
-  return posts.length ? (
+  return totalPosts > 0 ? (
     <InfiniteScroll
-      next={loadMoreItems}
+      next={async () => {
+        const newPosts = await getMorePosts({
+          contract: postStorageContracts[selectedType],
+          limitAmount: scrolledLimit,
+          loadedPostAmount: selectedPosts.length,
+        })
+        PostStore.posts[selectedType] = Promise.resolve([
+          ...selectedPosts,
+          ...newPosts,
+        ])
+        setScrolledLimit(PostStore.limit)
+      }}
       className={scrollContainer}
       style={{ overflow: 'hidden' }}
-      dataLength={loadedItemsAmount}
-      hasMore={loadedItemsAmount < posts.length}
+      dataLength={selectedPosts.length}
```

Comment:
You're using `selectedPosts.length` a couple of times, I think it's worth to extract it into `const` or destructure